docs/iris/example_code/General/custom_file_loading.py

```diff
@@ -63,74 +70,84 @@ def load_NAME_III(filename):
 
     """
 
-    # loading a file gives a generator of lines which can be progressed using the next() method.
-    # This will come in handy as we wish to progress through the file line by line.
-    file_handle = open(filename)
-
-    # define a dictionary which can hold the header metadata about this file
-    headers = {}
-
-    # skip the NAME header of the file which looks something like 'NAME III (version X.X.X)'
-    next(file_handle)
-
-    # read the next 16 lines of header information, putting the form "header name:    header value" into a dictionary
-    for _ in range(16):
-        header_name, header_value = file_handle.next().split(':')
-
-        # strip off any spurious space characters in the header name and value
-        header_name = header_name.strip()
-        header_value = header_value.strip()
-
-        # cast some headers into floats or integers if they match a given header name
-        if header_name in ['X grid origin', 'Y grid origin', 'X grid resolution', 'Y grid resolution']:
-            header_value = float(header_value)
-        elif header_name in ['X grid size', 'Y grid size', 'Number of fields']:
-            header_value = int(header_value)
-        elif header_name in ['Run time', 'Start of release', 'End of release']:
-            # convert the time to python datetimes
-            header_value = datetime.datetime.strptime(header_value, UTC_format)
-
-        headers[header_name] = header_value
-
-    # skip the next blank line in the file.
-    next(file_handle)
-
-    # Read the next 7 lines of column definitions
-    column_headings = {}
-    for column_header_name in ['species_category', 'species', 'cell_measure', 'quantity', 'unit', 'z_level', 'time']:
-        column_headings[column_header_name] = [col.strip() for col in file_handle.next().split(',')][:-1]
-
-    # convert the time to python datetimes
-    new_time_column_header = []
-    for i, t in enumerate(column_headings['time']):
-        # the first 4 columns aren't time at all, so don't convert them to datetimes
-        if i >= 4:
-            new_time_column_header.append(datetime.datetime.strptime(t, UTC_format))
-        else:
+    # Loading a file gives a generator of lines which can be progressed using
+    # the next() method. This will come in handy as we wish to progress through
+    # the file line by line.
+    with open(filename) as file_handle:
+        # Define a dictionary which can hold the header metadata for this file.
+        headers = {}
+
+        # Skip the NAME header of the file which looks something like
+        # 'NAME III (version X.X.X)'.
+        next(file_handle)
+
+        # Read the next 16 lines of header information, putting the form
+        # "header name:    header value" into a dictionary.
+        for _ in range(16):
+            header_name, header_value = file_handle.next().split(':')
```

Comment:
I see you've already [sorted this out](https://github.com/QuLogic/iris/tree/py3k-builtin-changes) in a [subsequent commit](https://github.com/QuLogic/iris/commit/067fd2e037bad20729f7e4d64f910add0b41ccf0). I'm perfectly happy to wait for that. :smile: